Auth48

cabo opened the pull request:
Files for completing AUTH48 process

cabo committed:
Auth48: Note c&p error; date

cabo committed:
Add 2024-02-07 RPC version of XML and extracted files; initial check

glyn reviewed (approved):
Not sure of the intended use of this. Is it mainly a base for applying XML changes? Is the content equivalent to the current level from the RFC editors? It looks like many of the files are generated from the XML, presumably just to give a more granular way of viewing changes.

Should we now create PRs relative to the auth48 branch? Not sure how to do that. Clearly, I can (at least in principle) edit the XML, but how to generate the rest?

cabo commented:
On 2024-02-12, at 15:41, Glyn Normington ***@***.***> wrote:
> 
> 
> @glyn approved this pull request.

Well, we aren’t done yet :-)

> Not sure of the intended use of this. Is it mainly a base for applying XML changes?

Yes.  None are in yet.

> Is the content equivalent to the current level from the RFC editors? It looks like many of the files are generated from the XML, presumably just to give a more granular way of viewing changes.

The other files are for check-abnf.

> Should we now create PRs relative to the auth48 branch? Not sure how to do that. Clearly, I can (at least in principle) edit the XML, but how to generate the rest?

I was planning to simply edit the rfc9535.xml based on the diff lists we now have; the other files are just helping to do consistency checks (they are created from the check-abnf script, but I committed them so we don’t overlook changes).

Grüße, Carsten

cabo committed:
First round of abnf spacing changes

cabo committed:
Put "; skip surrogate code points" on its own line

cabo committed:
Fix some weird blank-space inconsistencies in ABNF

cabo committed:
Use and reference "I-Regexp" term from RFC 9485

cabo committed:
Search and "at least one" substring

cabo committed:
Add DIGIT to name-first at the end

cabo committed:
(G27) Better Section Name "Function Extensions Subregistry"

cabo committed:
(G26) Use § 2.4.3 in both places for "well-typed" cross-reference

cabo committed:
(G25) Remove incorrect cross-reference in 2.3.1.1

(After correcting it to 2.3.1.1, it would become a meaningless self-reference.)

cabo committed:
(G21): Appendix B: Use JSONPath terminology for JSONPath

And clarify whose notation this sentence is about.

cabo committed:
(G18): Delete unnecessary text

(We use "evaluate" sparingly, and it is not needed at all here to
further qualify "produce".)

glyn commented on draft-ietf-jsonpath-base.md:
```diff
@@ -449,7 +449,7 @@ $.store.book[?@.price < 10].title
 | `..*`               | shorthand for `..[*]`                                                                                                   |
 | `'name'`            | [name selector](#name-selector): selects a named child of an object                                                     |
-| `*`                 | [wildcard selector](#name-selector): selects all children of a node                                                    |
+| `*`                 | [wildcard selector](#wildcard-selector): selects all children of a node                                            |
```
@cabo Good catch!

cabo committed:
(G10): Make sentence easier to understand

summing with -> adding to

cabo committed:
(G6): There is only one root identifier example

glyn reviewed (commented):
All changes up to G10 looking good, thanks!

cabo committed:
(G5): emphasize loneness by saying "containing a single node"

cabo committed:
(G2): Remove remnant from when the definition was for "argument"

note that we don't have a definition for a sole "argument", but that
is a standard term that has very little confusion attached to it.

cabo committed:
(G1): Remove unfortunate double use of the term "structure"

glyn reviewed (commented):
Another good set of changes, thanks, up to G1.

What became of G11?

glyn reviewed (approved):
All approved. Thanks!

glyn commented:
> Another good set of changes, thanks, up to G1.
> 
> What became of G11?

I dropped G11 after discussion.

cabo committed:
(C3): Avoid <referencegroup XML that is still at risk.

gregsdennis commented:
@cabo do you have a link for the auth48 process and what these files are for?  I thought auth48 was just final author approvals.

cabo commented:
Hi Greg,

there is not a single document I could point to.
GPT-4 says:

> The Auth48 process for IETF standards is specified in several documents, but the main ones are:

> RFC 2026: The Internet Standards Process -- Revision 3. This document covers the general standards process at IETF, including the final AUTH48 review state when a draft becomes a full Internet Standard.
> The IETF's official "Guidelines to Authors of Internet-Drafts", also called "Guidelines to Authors."
> The IETF's official "Guidelines to Working Group Chairs," which also mentions the AUTH48 process.
> In addition, the IETF Trust's "Legal Provisions Relating to IETF Documents" include guidelines for authors during the AUTH48 review period.

In summary, AUTH48 is for checking the RFC Production Center (RPC) edits between authors and RPC and making any editorial and technical changes necessary before publication.  Since this is an approved document, any technical changes will be quite limited (essentially inconsequential for interoperability in this case).

gregsdennis commented:
So why all the new files?  What's the driver for these?

I don't expect any technical changes, but I'm curious about the new files.

cabo commented:
The new files come from a script, check-abnf, that just checks whether any of the spacing fixes we made to the ABNF actually changed its contents.  They are extracted from the XML via kramdown-rfc-extract-sourcecode.  I was lazy and comitted all of those, even if I only care about the ABNF.

gregsdennis reviewed (approved):
Thanks for the explanation, @cabo

cabo committed:
RPC changes communicated at 2024-02-13 23:22Z

glyn reviewed (approved):
5c10c5a737b25d434aea2f4d07e2f6d04f9ba104 LGTM, thanks.

cabo committed:
(C4) Complete fix for placement of "at most".

cabo committed:
(C5): Remove overzealous duplicate redundant superfluous citation

glyn commented on auth48/rfc9535.xml:
```diff
@@ -72,7 +72,7 @@
       <t>JSON <xref target="RFC8259"/> is a popular representation
 format for structured data values.
-JSONPath defines a string syntax for selecting and extracting JSON <xref target="RFC8259"/> values
+JSONPath defines a string syntax for selecting and extracting JSON values
```
Yes, that was a bit much!

glyn reviewed (approved):
C4 and C5 LGTM, thanks!

cabo committed:
(C6) simplify editorial memento for ES4

(We can't verify what was deprecated in 2014, which was 5 years after
ES5 was released, replacing the ES4 proposal stalled in development.)

cabo committed:
(C1) 3 more word-joiners to avoid confusing line breaks

cabo committed:
(C7) use past tense for IANA registration that has happened

glyn reviewed (approved):
C6, C1, C7 are good.

cabo committed:
(C8) Avoid superfluous "strict" in table 11, twice

The word "strict" is used nowhere else but table 11.
It serves no useful purpose here.
(It might have been meant to distinguish strict inequalities (<) from
non-strict (<=) ones, but extracting that intention is too much of a
transfer for the reader.)

glyn reviewed (approved):
C8 is just what I would have done.

cabo committed:
(C9) We now distinguish empty nodelists from Nothing

The text in 2.4.8 (value()) apparently wasn't updated when we stopped
talking about "Nothing" for nodelists (these are now simply "empty").

glyn reviewed (approved):
C9 was a corker. Thanks for fixing.

cabo commented:
I'm now planning to keep this as a draft pull request until the RFC is published.
Then I'll have a look if the published version can be back-ported to the .md
